### DISCUSSION TITLE
Property is setting as ObjectId instead of an Object

### DISCUSSION THREAD

**Discussion by Reh4m** [typegoose/typegoose#965] [Q&A] [answered]
Hi guys, I got a GraphQL API with type-graphql and typegoose. Two properties of my Order collection are stored as ObjectId and not as a object and an object array respectively.

![image](https://github.com/user-attachments/assets/7bfeadfc-9f72-46e5-af84-1df590a1b611)

For this, I have two classes (of type @ObjectType, not collections), Address and OrderDetails. These classes define the properties that they must have and then validate using class-validator. I don't want to store them as a collection with an ObjectId, only as a simple object and an object array (of type Address and OrderDetails).

```javascript
// Order Model

@Field(() => Address)
@Property({ type: Address, required: true })
shipAddress: Address;

@Field(() => [OrderDetails])
@Property({ type: () => [OrderDetails], required: true })
details: OrderDetails[];
```

```javascript
@ObjectType()
export class Address {
  @Field((type) => String)
  contactName: string;

  @Field((type) => String)
  contactPhone: string;

  @Field((type) => String)
  address: string;

  // ...
}

```

```javascript
@ObjectType()
export class OrderDetails {
  @Field((type) => Product)
  product: Product;

  @Field((type) => Number)
  unitPrice: number;

  @Field((type) => Number)
  quantity: number;

  @Field((type) => Number)
  discount: number;
}
```

I hope someone can help me. Thanks.

**Comment by hasezoey** [comment, marked as answer]
> Two properties of my Order collection are stored as ObjectId and not as a object and an object array respectively.

The formatting from Compass might be a bit confusing, but if you look closely, `shipAddress` is a object and `details` is a object array. As for the `_id` property, mongoose automatically adds those, even if they are subdocuments. You can disable this via `@prop` option [`_id: false`](https://typegoose.github.io/typegoose/docs/api/decorators/prop/#_id) or via [`@modelOptions({ schemaOptions: { _id: false } })`](https://mongoosejs.com/docs/guide.html#_id).

As for why no other fields are showing up, this is because none of those fields are marked `@prop`, which is required for typegoose to see them, and even if you do that, if they are not marked `@prop({ required: true })`, then they also dont show up unless specified.

**Reply by Reh4m** [reply]
That makes sense. So, every property to be stored in mongo must have `prop` to show up. Even if it's not a collection. If I'm not mistaken.

**Reply by hasezoey** [reply]
> So, every property to be stored in mongo must have prop to show up

Yes that is correct, also see the documentation note:

> Any Property that does not have @prop on it will not be in the Schema, because typegoose's current approach is that the keys for the schema get defined in @prop, unlike class-transformer's global-@Expose and @Exclude.

[Source](https://typegoose.github.io/typegoose/docs/api/decorators/prop/)

**Reply by Reh4m** [reply]
Thank you! It's good to know that there're someone answering this questions.
